app/src/main/scala/org/alephium/explorer/persistence/dao/UnconfirmedTxDao.scala

```diff
@@ -99,7 +103,35 @@ object UnconfirmedTxDao extends UnconfirmedTxDao {
           tx.hash,
           tx.chainFrom,
           tx.chainTo,
-          inputs.filter(_.txHash == tx.hash).map(_.toApi),
+          inputs.filter(_.txHash == tx.hash).sortBy(_.uinputOrder).map(_.toApi),
+          outputs.filter(_.txHash == tx.hash).sortBy(_.uoutputOrder).map(_.toApi),
+          tx.gasAmount,
+          tx.gasPrice,
+          tx.lastSeen
+        )
+      }
+    })
+  }
+
+  def listByP2PKHAddress(address: Address)(
+      implicit executionContext: ExecutionContext,
+      databaseConfig: DatabaseConfig[PostgresProfile]): Future[Seq[UnconfirmedTransaction]] = {
+    run(for {
+      txHashes <- UInputSchema.table
+        .filter(_.p2pkhAddress === address)
+        .map(_.txHash)
+        .distinct
+        .result
+      txs     <- UnconfirmedTxSchema.table.filter(_.hash inSet txHashes).sortBy(_.lastSeen.desc).result
+      inputs  <- UInputSchema.table.filter(_.txHash inSet txHashes).result
+      outputs <- UOutputSchema.table.filter(_.txHash inSet txHashes).result
+    } yield {
```

Comment:
Hey Thomas, just checking if we are working on reducing the number of concurrent queries dispatched? To avoid the incident from last week of high network IOps. 

From the looks of it, these queries could be a single query using `OUTER JOIN` etc 

Similarly in your last PR [here](https://github.com/alephium/explorer-backend/pull/320/files#diff-25723e7a7dcf656a9cecf4b53ce0e8ae1468880aaa4772ecfe503bd7b771886cR90-R95), these could also be written in a single query. Bringing this up now, in case these optimisations are being deferred to another issue.

Reply:
I will need to benchmark all these, because one of our goal was to avoid heavy `JOIN` as they are slow. 
But I agree IOps definitively needs to be considered in the start now, thx for raising that :+1:

Reply:
I'm thinking a bit more on the subject, for this case of unconfirmed transactions, where we don't do complicated queries and joining. We could maybe actually store the serialization of the tx as a byte array in the `UnconfirmedTxSchema`, this would look a bit like a key value store. 

Any opinion on this?

Reply:
I gave a try [here](https://github.com/alephium/explorer-backend/commit/91974b22ef671120065a999224db4e39da8ad7c8)

Reply:
> I gave a try [here](https://github.com/alephium/explorer-backend/commit/91974b22ef671120065a999224db4e39da8ad7c8)

Nice 👍🏼. Left a small comment. 

>  We could maybe actually store the serialization of the tx as a byte array in the `UnconfirmedTxSchema`, this would look a bit like a key value store.
>
> Any opinion on this?

Oh we do something similar with `AppStateSchema` right. 

My *opinion*, I'm not sure. But it be nice if the data we create is laid out the way the database we use likes it laid out as it's optimised for it. And incase we do wanna query the table in the future for some admin UI and display analytics. But thats just *opinion* :) I'm sure there are good reasons for it to be a key-value table.

Reply:
yes I agree we still need to keep the data "queryable", the idea is not to have dynamic values, as for a tx_id the data is immutable. In this  `UnconfirmedTx` case, we still store all data the sql way, so we can make more complex queries in the future, but storing the full transaction at the same time allows to avoid JOINS when not necessary....But I'm still not sure too  ....maybe it's not worth it for the `UnconfirmedTx` as the table stays small.

Reply:
In the end I went first with a solution following what we have in our `TransactionQueries`, using only raw sql to improve perf.
I'm not sure we can nicely joins the 3 tables, as inputs and outputs are really 2 different things and don't need to be joined.

Reply:
Oh it looked like three queries (3 DB trips) running on the same input so a join should be possible. But maybe I'm overlooking something. Please mark this as resolved if you are ok with it.

Reply:
the problem I see:
first we join the tx table with the input on the `tx_hash`, let say we have 3 inputs for one tx: the output rows will be:
```
tx, in1
tx, in2,
tx, in3
```
now we need to join to get the outputs also on the `tx_hash`, let say 2 outputs:
```
tx, in1, out1
tx, in1, out2
tx, in2, out1
tx, in2, out2
tx, in3, out1
tx, in3, out3
```
Am I right?

We unnecessary return multiple time the inputs and outputs, while we only want 1 tx, 3 inputs and 2 outputs. Imagine a tx with 100 inputs and 100 outputs.

But maybe I miss something? or maybe it's not an issue compared to querying 3 times the DB?

Reply:
You are right. It would return duplicate rows. Which is ok. But why would there be a need to read 100 inputs and 100 outputs in a single query anyway? Following your example, that would be 10,000 rows? We shouldn't be reading that much data in a single query anyway unless it's for aggregation/analytics which can be done within the query and deferred to the DB. Do we have a UI that displays that much data? I'm trying to look for where `UnconfirmedTxDao.list` and `listByP2PKHAddress` are being called, I couldn't find anything.

Reply:
Thomas I will leave this to your judgement. I'm sure this is well thought out. Don't think I'm being helpful here, I'm just raising more questions. I can always give it a performance check later, if possible.

Reply:
if someone is requesting the details of such a tx, we have to return every inputs and outputs for that given tx, we can't afford to skip some of them otherwise if you sum up everything you will think that you miss some ALPH. Knowing that all those transactions endpoints are also called from the wallet, it would be a problem.
And yes in the explorer we will display all the inputs and outputs in the transaction page

Reply:
your questions are always good and helps me a lot improving the perfs, in this particular case it gave me the opportunity to really think of a "one query thing". I just couldn't find a good one fitting our requirement, maybe there is....